include/cursor.h

```diff
@@ -0,0 +1,85 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+#ifndef __LABWC_CURSOR_H
+#define __LABWC_CURSOR_H
+
+#include <wlr/util/edges.h>
+
+struct seat;
+struct server;
+
+/* Cursors used internally by labwc */
+enum lab_cursors {
+	LAB_CURSOR_CLIENT = 0,
+	LAB_CURSOR_DEFAULT,
+	LAB_CURSOR_GRAB,
+	LAB_CURSOR_RESIZE_NW,
+	LAB_CURSOR_RESIZE_N,
+	LAB_CURSOR_RESIZE_NE,
+	LAB_CURSOR_RESIZE_E,
+	LAB_CURSOR_RESIZE_SE,
+	LAB_CURSOR_RESIZE_S,
+	LAB_CURSOR_RESIZE_SW,
+	LAB_CURSOR_RESIZE_W
+};
+
+#ifdef __LABWC_IN_CURSOR_C
```

Comment:
I'm not a huge fan of the #ifdef here.  Maybe just put these in `cursor.c`, and leave a comment, "If you add to this enum, remember to update cursors_xdg and cursors_x11 in cursor.c"?

Reply:
Yeah.. I'd like to have all the 3 in a single place to make it harder for people to shoot themselves in the foot in the future but I am also not much a fan of that `ifdef` include stuff either.

Trade-off seams to be somewhat along the lines of "don't do something unexpected" and "make sure those structs are always kept in sync". 
Guess a comment on both ends would do as well.

Reply:
Do we have access to C11 `static_assert`?  If so, then we could add a final `NUM_LAB_CURSORS` value to the enum and then `static_assert` that the length of the arrays are equal to `NUM_LAB_CURSORS`.

Reply:
I think so
https://github.com/labwc/labwc/blob/89890b6be9bb79b701b5b257eb8efee0204ec744/meson.build#L8

It’s a nice idea. A bit like linux BUG_ON() macro?

Reply:
Nice idea. This seems to work fine:
```c
#define ARRAY_SIZE(array) (sizeof(array) / sizeof((array)[0]))
static_assert(ARRAY_SIZE(cursors_xdg) == LAB_CURSOR_COUNT, "XDG cursor names are out ot sync");
static_assert(ARRAY_SIZE(cursors_x11) == LAB_CURSOR_COUNT, "X11 cursor names are out ot sync");
#undef ARRAY_SIZE
```